clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -497,24 +490,57 @@ protected CreateServiceRequest makeServiceRequest(
 
     CreateServiceRequest request =
         new CreateServiceRequest()
-            .withServiceName(newServerGroupName)
+            .withServiceName(newServerGroupName.getServiceName())
             .withDesiredCount(desiredCount)
             .withCluster(description.getEcsClusterName())
-            .withLoadBalancers(loadBalancers)
             .withTaskDefinition(taskDefinitionArn)
             .withPlacementConstraints(description.getPlacementConstraints())
             .withPlacementStrategy(description.getPlacementStrategySequence())
             .withServiceRegistries(serviceRegistries)
             .withDeploymentConfiguration(deploymentConfiguration);
 
+    List<Tag> taskDefTags = new LinkedList<>();
     if (description.getTags() != null && !description.getTags().isEmpty()) {
-      Collection<Tag> taskDefTags = new LinkedList<>();
       for (Map.Entry<String, String> entry : description.getTags().entrySet()) {
         taskDefTags.add(new Tag().withKey(entry.getKey()).withValue(entry.getValue()));
       }
-      request.withTags(taskDefTags).withEnableECSManagedTags(true).withPropagateTags("SERVICE");
     }
 
+    // Apply moniker strategy which may add tags
+    namer.applyMoniker(
+        new EcsResource() {
+          @Override
+          public String getName() {
+            return request.getServiceName();
+          }
+
+          // Used by Frigga when moniker support is disabled
+          public void setName(String name) {
+            request.setServiceName(name);
+          }
+
+          @Override
+          public List<Tag> getResourceTags() {
+            return taskDefTags;
+          }
+        },
+        newServerGroupName.getMoniker());
+
+    // Only add tags if they're set as it's an optional feature for ECS
+    if (taggingEnabled) {
+      request
+          .withTags(taskDefTags)
+          .withEnableECSManagedTags(true)
+          .withPropagateTags(PropagateTags.SERVICE.toString());
+    } else {
+      if (!taskDefTags.isEmpty()) {
+        throw new IllegalArgumentException(
+            "Tagging is not enabled on ECS account " + description.getAccount() + ".");
```

Comment:
Can you make this exception a bit more descriptive? Users might not be aware that they can't tag without enabling the new (long) arn format.

Reply:
> ECS account settings for account Test do not allow tagging as `serviceLongArnFormat` and `taskLongArnFormat` are not enabled.

How about that?

Reply:
Looks great! Thanks! 💯